Migrate the site infrastructure to Jaspr and Jaspr Content

parlough opened the pull request:
**Staged:** https://flutter-docs-prod--pr12548-feat-jaspr-migration-kz7c0jfj.web.app/

Migrates the site from being built with JS/TS, Node, and 11ty to being built with purely Dart, [Jaspr](https://jaspr.site/), and [Jaspr Content](https://docs.jaspr.site/content).

---

As part of this work, the `dash_site` bash script has been removed from the root of the repository.

After the removal, you can access the site tooling, including building and staging with `dart run dash_site <command>` from anywhere within the repository workspace. If you still want a shorter command, you can add an alias for `dart run dash_site` in your shell, such as `dst`.

---

**Staged:** https://flutter-docs-prod--pr12548-feat-jaspr-migration-kz7c0jfj.web.app/

Resolves https://github.com/flutter/website/issues/12405
Fixes https://github.com/flutter/website/issues/12590

parlough commented:
@schultek Can you take a look at why the `llms.txt` file is being rendered as HTML rather than being passthrough rendered? Perhaps I configured the `rawOutputPattern` incorrectly or misunderstood it. I appreciate the help!

ericwindmill reviewed (approved):
LGTM, with two asterisks

*I'm approving based on the code, independently of clicking through the staged site.
*I really only skimmed through it. From a high-level, it looks great, and I generally trust you two and don't feel the need to scrutinize every line. LMK if there's anything that you feel specifically you'd like eyeballs on. 

Exciting!

parlough commented:
Thank you both for taking a look and the feedback!

Addressing each of the points raised by @sfshaza2 with resolutions and a few questions:

> On the front page, the What's New in Flutter video is smaller on the staged site. Can we restore the OG size?

I've fixed this! Now the `YouTubeEmbed` component supports a `fullWidth` attribute that I've also added to that embed :D 

> There is additional white space on the left side (following the sidenav) on new pages that are missing the On this page column from the new page. [OG page](https://docs.flutter.dev/ui/widgets/material). [New page](https://flutter-docs-prod--pr12548-feat-jaspr-migration-kz7c0jfj.web.app/ui/widgets/material). Other pages do have the On this page column, so did I happen to select one where it's turned off?

The widget catalog pages happen to be pages where I turned off the TOC in this PR, so I believe this difference is to be expected. If you notice it elsewhere though, do please let me know :)

> OK, hang in there. Expand the Testing & debugging menu in the sidenav. Then, under Testing, select Overview. It brings up the page. Select Overview again. And again. On the OG page, things seem good. On the new page, every time, you select Overview, the sidenav jumps (as if it's missing a test to see if the page is already up).

Nice catch. Fixed! Scrolling to the active entry in the sidenav should happen before page load finalizes now.

> On the new site (on the same page as in the previous bullet), there are extra blank lines in the top two bulleted lists, but not in the last bulleted list. A rather weird inconsistency.

I see this inconsistent spacing on the [old/live version](https://docs.flutter.dev/testing/overview) but not on the [new version](https://flutter-docs-prod--pr12548-feat-jaspr-migration-kz7c0jfj.web.app/testing/overview). Does that match what you're seeing? If there's still an issue though, can you send a screenshot?

> There seems to be a difference in font. [OG page](https://docs.flutter.dev/testing/errors). [New page](https://flutter-docs-prod--pr12548-feat-jaspr-migration-kz7c0jfj.web.app/testing/errors). Scroll to the Handling all types of headers header on each pages. The font is thinner and narrower on the new site, making it harder to read.

I can't seem to reproduce this. The fonts look the same when I overlay the pages. Can you try hard refreshing each page and let me know if you still notice differences? And send a screenshot of both. Thanks!

> Have we run the updated pages through accessibility testing?

Good idea, thanks! I went ahead and did so and I fixed the one regression that came up as well as fixed a pre-existing issue.

parlough commented on site/lib/src/client/global_scripts.dart:
```diff
@@ -0,0 +1,427 @@
+  for (var i = 0; i < headings.length; i++) {
+    observer.observe(headings.item(i) as web.Element);
+  }
+}
```
Great question. I or Kilian can go back and add some details in comments about why each of these is here.

It depends on the function, but I believe there are three primary reasons:

- Some are hopefully temporary until Jaspr supports "server components" where client components can accept server-only components as arguments. This is mostly to avoid the client boundary being unnecessary high in the tree for now, but technically could be tied to a Jaspr component today if we wanted.

  _Kind of related, but https://dev.to/this-is-learning/why-efficient-hydration-in-javascript-frameworks-is-so-challenging-1ca3 (by the author of SolidJS) was a helpful read for me as I worked to understand hydration, server components, data serialization, etc. I recommend it!_

- Sometimes imperative DOM logic is much simpler.

- Some of the functions are for features or logic we will hopefully supersede soon and directly migrating the old JS to Dart is easier while still being a DevEx win for us.

sfshaza2 reviewed (approved):
lgtm
